### PR TITLE
Readd 'Add Layer' link to map edit dropdown in header

### PR DESCRIPTION
**Pull request by CloudNiner**

### CODE REVIEW

File: lib/assets/javascripts/cartodb/table/header/options_menu.js [hunk 1]
```diff
@@ -42,6 +43,7 @@ cdb.admin.HeaderOptionsMenu = cdb.admin.DropdownMenu.extend({
     this.user = this.options.user;
     this.dataLayer = this.options.dataLayer;
     this.table = this.dataLayer.table;
+    this.vis = this.options.vis;
```
Comment (rmartz) [inline lib/assets/javascripts/cartodb/table/header/options_menu.js:46]: Seems like we don't use this, later on we refer to it by `this.options.vis`

Comment (CloudNiner) [inline lib/assets/javascripts/cartodb/table/header/options_menu.js:46]: Whoops. I'll update the new functions to use `this.vis`. It helps to see at a glance what the required options are for the view.

**Comment by rmartz** [comment]
I'm running into a problem that adding a layer doesn't insert it into the sidebar until you refresh, not sure if that's new to this PR or not.

**Comment by CloudNiner** [comment]
Any console errors? It seems to work fine for me, with both remote datasets and local.

**Comment by rmartz** [comment]
Nothing new, just a couple failing GET requests that seem unrelated (One is for something on `hs-analytics.net`, another is for `noise.png`). Is there a command I need to run to get everything up to latest? I've been using `grunt js && sudo service cartodb_server restart`

**Comment by CloudNiner** [comment]
Alright, still unable to reproduce this. I'm guessing its a side effect of not having FDW setup on your box. I'm bringing up a box with fdw disabled to test, in the meantime going to merge this so I can continue blocked work.

**Pull request merged**